lib/src/config_delayed_merge.cc

```diff
@@ -23,8 +23,8 @@ namespace hocon {
         throw config_exception("called value_type() on value with unresolved substitutions, need to config#resolve() first, see API docs");
     }
 
-    vector<shared_value> const& config_delayed_merge::unmerged_values() const {
-        return _stack;
+    shared_ptr<vector<shared_value>> config_delayed_merge::unmerged_values() const {
+        return make_shared<vector<shared_value>>(_stack);
```

Comment:
This copies _stack. I was thinking we should represent _stack internally as a shared_ptr to vector. unmerged_values should return `shared_ptr<const vector<shared_value>>`.